pkg/sql/sem/tree/expr.go

```diff
@@ -885,7 +885,7 @@ func (node *Array) Format(ctx *FmtCtx) {
 	// UNKNOWN[], since that's not a valid annotation.
 	if ctx.HasFlags(FmtParsable) && node.typ != nil {
 		if node.typ.ArrayContents().Family() != types.UnknownFamily {
-			ctx.WriteString(":::")
+			ctx.WriteString("::")
```

Comment:
i'm wondering if what we need is to keep track of whether there already was a type annotation wrapping the `Array` expr. or alternatively, update `AnnotateTypeExpr.Format` so it doesn't print the annotation if the inner expr already is going to